tests/test-ellama.el

```diff
@@ -0,0 +1,52 @@
+;;; test-ellama.el --- Ellama tests -*- lexical-binding: t -*-
+
+;; Copyright (C) 2023  Free Software Foundation, Inc.
+
+;; Author: Sergey Kostyaev <sskostyaev@gmail.com>
+
+;; This file is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation; either version 3, or (at your option)
+;; any later version.
+
+;; This file is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;
+;; Ellama tests.
+;;
+
+;;; Code:
+
+(require 'cl-lib)
+(require 'ellama)
+(require 'ert)
+
+(ert-deftest test-ellama--code-filter ()
+  (should (equal "" (ellama--code-filter "")))
+  (should (equal "(hello)" (ellama--code-filter "(hello)")))
+  (should (equal "(hello)" (ellama--code-filter "```lisp\n(hello)```"))))
+
+(ert-deftest test-ellama-code-improve ()
+  (let ((original "(hello)")
+        (improved "```lisp\n(hello)```"))
```

Comment:
Maybe
```emacs-lisp
(improved "```lisp\n(hello)\n```"))
````
?

Reply:
Yes, true. I added a newline to the code snippet example generated by the LLM